handle.go

```diff
@@ -65,22 +65,37 @@ func NewHandle(rules any, keyObfuscatorRegex string, valueObfuscatorRegex string
 
 	config := newConfig(&encoder.cgoRefs, keyObfuscatorRegex, valueObfuscatorRegex)
 	diagnosticsWafObj := new(wafObject)
+	defer func() {
+		if !diagnosticsWafObj.isInvalid() {
+			wafLib.wafObjectFree(diagnosticsWafObj)
+		}
+	}()
+
+	cHandle := wafLib.wafInit(obj, config, diagnosticsWafObj, encoder.cgoRefs)
+	// Upon failure, the WAF may have produced some diagnostics to help signal what went wrong...
+	var diags *Diagnostics
+	if !diagnosticsWafObj.isInvalid() {
+		diags, err = decodeDiagnostics(diagnosticsWafObj)
+	}
 
-	cHandle := wafLib.wafInit(obj, config, diagnosticsWafObj)
-	keepAlive(encoder.cgoRefs)
-	// Note that the encoded obj was copied by libddwaf, so we don't need to keep them alive
-	// for the lifetime of the handle (ddwaf API guarantee).
 	if cHandle == 0 {
+		// WAF Failed initialization, report the best possible error...
+		if err == nil && diags != nil {
```

Comment:
`err` is not clearly coming from line 78 to me when `!diagnosticsWafObj.isInvalid()` is false.
I think you can safely remove the `err == nil` part of the condition because `diags` is `nil` by default:
```suggestion
		if diags != nil {
```